modules/configure_repo/main.ruleset_main.tf

```diff
@@ -0,0 +1,29 @@
+resource "github_repository_ruleset" "main" {
+  name        = "main"
+  repository  = data.github_repository.this.name
+  target      = "branch"
+  enforcement = "active"
+
+  conditions {
+    ref_name {
+      include = ["~DEFAULT_BRANCH"]
+      exclude = []
+    }
+  }
+
+  rules {
+
+    creation                = true #allow no-one to create main?
+    deletion                = true #restrict deletion of main to users with bypass (no-one)?
+    required_linear_history = true
+    non_fast_forward = true #no force pushes
+
+    pull_request {
+      dismiss_stale_reviews_on_push = true
+      require_code_owner_review = true 
+      required_approving_review_count = 0      
```

Comment:
We need a way of making this a `1` for specific repos.

As we have some ALZ compliance requirements to other PG teams that need this to be a `1` so they can use ALZ